### ISSUE TITLE
problem about  code

### ISSUE THREAD

**Issue by Devin147** [takiyu/portrait_matting#6]
# Increase gradient of the probability
        prob = F.softmax(score, axis=1)
        prob = prob - 0.5
        prob = self.prob_scale(prob)
        prob = F.clip(prob, -0.5, 0.5)
        prob = prob + 0.5
        #in  __init__():
        self.prob_scale = L.Scale(W_shape=(1,))
        self.prob_scale.W.copydata(np.array([100.0]))

         why you scale prob 100?

**Comment by Raj-08** [comment]
Why do you think its a problem @Devin147 ?

**Comment by GuideWsp** [comment]
it is wrong. when you scale the prob, so after clip, all the prob are zero. i have check it. so i am sure it was wrong

**Comment by dsArtem** [comment]
@GuideWsp could you share solution to this problem. If it was found.

**Comment by GuideWsp** [comment]
I change it as :
    prob = F.softmax(score, axis=1)
    prob = prob - 0.5
    prob = F.clip(prob, -0.5, 0.5)
    prob = prob + 0.5

finaly, i have to abandon this algorithm because i really can't produce the result as the paper show.  i email to the author, but got no answers.

**Comment by dsArtem** [comment]
Ok, thanks. I will try to fix it. Or.... Did you find other realizations of matting?

**Comment by GuideWsp** [comment]
you can try "Deep Image Matting". if you give a relative good trimap, it will give a good results. but it isn't an automatic algorithm.  i nearly give up this topic for som reasons. recently, i focus on semantic segmentation and the related topics.

**Comment by dsArtem** [comment]
I found the problem. This is due to the initialization of weights. 
fcn8s.py:
    kwargs = {
               'initialW': chainer.initializers.Zeros(),
                'initial_bias': chainer.initializers.Zerosl(),
            }
fixed:
    kwargs = {
                'initialW': chainer.initializers.Normal(),
                'initial_bias': chainer.initializers.Normal(),
            }

**Comment by GuideWsp** [comment]
maybe it's right. can you tell me when you get the results as the paper shown? thanks

**Comment by GuideWsp** [comment]
if possible, you can add my wechat "sky__wsp", then we can talk about this.@dsArtem

**Comment by dsArtem** [comment]
Ok. I'll let you know when I get the results.

**Comment by assertdebug** [comment]
How to fixed this problem ? 
self.prob_scale.W.copydata(np.array([100.0])) @dsArtem

**Comment by puphy** [comment]
@dsArtem Hi, do you get a good result? I just cant't make the training convergence...

**Comment by kkkmax** [comment]
Brother, my three modes of training:
$ python scripts/train.py --mode seg
$ python scripts/train.py --mode seg+
$ python scripts/train.py --mode seg_tri

The training part looks normal, but the test was very bad and found:
    mode=seg
    mode=seg+
The test result is a gray picture

   mode = seg_tri
The test result is a white picture

My test command is:
$ python scripts/train.py --mode {seg,seg+,seg_tri} --model_path './result/model_best' -i <image path>

Can you see where I am using the error? such as:
1) Is the model path I loaded correct?
2) Like you said W B needs to be changed to standardized initialization
3) Testing or training There is a particularly critical detail to note?

If you see a reply, I will be very grateful and happy!

@takiyu 
@dsArtem
@Raj-08
@Devin147 
@GuideWsp  
@assertdebug

**Comment by headhu** [comment]
Hey @dsArtem is there a way to reach you by email? I think we could share some knowledge regarding this subject

**Comment by Maru-chan511** [comment]
Does anybody get the results as the paper shown?

**Comment by Maru-chan511** [comment]
@kkkmax why don't you try the last mode "mat"?
